.evergreen/scripts/find-cmake-version.sh

```diff
@@ -0,0 +1,240 @@
+# attempt to install a version of CMake.
+cmake_replace_latest() {
+  declare -r cache="${1:?"missing path to cache directory"}"
+  declare -r root="${2:?"missing path to latest CMake root directory"}"
+  declare -r version="${3:?"missing latest CMake version"}"
+  declare -r cmake_cache="${cache}/cmake-${version}"
+
+  # Doesn't matter who creates the cache directory.
+  mkdir -p "${cache}" || return
+
+  # Should not happen, but handle it anyways.
+  [[ -L "${cmake_cache}" ]] || rm -rf "${cmake_cache}" || return
+
+  # Temporary directory to create new symlink.
+  declare tmpdir
+  tmpdir="$(make_tmpdir_in "${cache}")" || return
+
+  # Symlink to the new CMake root directory.
+  ln -s "${root}" "${tmpdir}/cmake-${version}" || return
+
+  # Atomic substitution of symlink from old to new CMake root directory.
+  mv -f "${tmpdir}/cmake-${version}" "${cache}" || return
+
+  # Be nice and clean up the now-empty temporary directory.
+  rmdir "${tmpdir}" || return
+}
+
+# find_cmake_version
+#
+# Usage:
+#   find_cmake_version 3 1 0
+#   cmake_binary="$(find_cmake_version 3 1 0)"
+#   cmake_binary="$(find_cmake_version 3 25 0 2>/dev/null)"
+#
+# Return 0 (true) if a CMake binary matching the requested version is available.
+# Return a non-zero (false) value otherwise.
+#
+# If successful, print the name of the CMake binary to stdout (pipe 1).
+# Otherwise, no output is printed to stdout (pipe 1).
+#
+# Diagnostic messages may be printed to stderr (pipe 2). Redirect to /dev/null
+# with `2>/dev/null` to silence these messages.
+#
+# Example:
+#   cmake_binary="$(find_cmake_version 3 25 0)" || exit
+#   "${cmake_binary:?}" -S path-to-source -B path-to-build
+find_cmake_version() {
+  declare -r major="${1:?"missing CMake major version"}"
+  declare -r minor="${2:?"missing CMake minor version"}"
+  declare -r patch="${3:?"missing CMake patch version"}"
+  declare -r version="${major}.${minor}.${patch}"
+
+  declare -r cache_dir="$HOME/.cache/mongo-c-driver"
+  declare -r cmake_cache_dir="${cache_dir}/cmake-${version}"
+  declare -r cmake_binary="${cmake_cache_dir}/bin/cmake"
+
+  if command -v "${cmake_binary}" >/dev/null; then
+    if "${cmake_binary}" --version | grep -q "${version}"; then
+      echo "Found CMake ${version} in mongo-c-driver CMake cache" 1>&2
+      printf "%s" "${cmake_binary}"
+      return # No work to be done: required CMake binary already exists in cache.
+    fi
+
+    # Shouldn't happen, but log if it does.
+    echo "Found inconsistent CMake version in mongo-c-driver CMake cache" 1>&2
+  fi
+
+  declare platform
+  declare extension
+  declare decompressor
+  declare decompressor_args
+  declare root_dir
+
+  # This logic is only applicable to CMake 3.20 and newer. For CMake versions
+  # earlier than 3.20 or CMake versions without prebuild binaries for certain
+  # platforms fallback to building from source.
+  case "${OSTYPE:?}-${HOSTTYPE:?}" in
```

Comment:
I was able to get away with a more concise method of download+extract, which may be useful: https://github.com/mongodb/libmongocrypt/blob/1d89e986fe682af4b3b8244014f2a6b62dc0b7b0/.evergreen/init.sh#L235-L256

AFAIK, all versions of `tar` that we use will support the features I'm using. (e.g. `--stip-components`, inferring the compression type from the magic number if it is passed a file directly with `-f` (as opposed to stdin)).

Reply:
Fair. I could go with either shape.